### PR TITLE
Dexfinance adapter

### PR DESCRIPTION
**Pull request by tuco-gordian**
Hi! Where can I store apiKey for requests to the graph?

### CODE REVIEW

**Comment by llamatester** [comment]
The dexswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators dexswap
    
    🦙 Running DEXSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 18 Jul 2024 11:23:00 GMT
    End Date:	Fri, 19 Jul 2024 11:23:00 GMT
    ---------------------------------------------------
    
    ClientError: auth error: missing bearer token: {"response":{"errors":[{"message":"auth error: missing bearer token"}],"status":200,"headers":{}},"request":{"query":"{\n\t\t\tdexSwapFactories {\n\t\t\t\ttotalVolumeETH\n\t\t\t}\n\t\t}"}}
        at /home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/src/index.ts:498:11
        at step (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:67:23)
        at Object.next (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:48:53)
        at fulfilled (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:39:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5) {
      response: {
        errors: [ [Object] ],
        status: 200,
        headers: Headers { [Symbol(map)]: [Object: null prototype] }
      },
      request: {
        query: '{\n\t\t\tdexSwapFactories {\n\t\t\t\ttotalVolumeETH\n\t\t\t}\n\t\t}',
        variables: undefined
      },
      chain: 'arbitrum'
    }

**Comment by llamatester** [comment]
The dexswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs dexswap
    
    🦙 Running DEXSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 18 Jul 2024 11:24:07 GMT
    End Date:	Fri, 19 Jul 2024 11:24:07 GMT
    ---------------------------------------------------
    
    ClientError: auth error: missing bearer token: {"response":{"errors":[{"message":"auth error: missing bearer token"}],"status":200,"headers":{}},"request":{"query":"{\n\t\t\tdexSwapFactories {\n\t\t\t\ttotalVolumeETH\n\t\t\t}\n\t\t}"}}
        at /home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/src/index.ts:498:11
        at step (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:67:23)
        at Object.next (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:48:53)
        at fulfilled (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:39:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5) {
      response: {
        errors: [ [Object] ],
        status: 200,
        headers: Headers { [Symbol(map)]: [Object: null prototype] }
      },
      request: {
        query: '{\n\t\t\tdexSwapFactories {\n\t\t\t\ttotalVolumeETH\n\t\t\t}\n\t\t}',
        variables: undefined
      },
      chain: 'arbitrum'
    }

**Comment by llamatester** [comment]
The dexswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees dexswap
    
    🦙 Running DEXSWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 18 Jul 2024 11:25:12 GMT
    End Date:	Fri, 19 Jul 2024 11:25:12 GMT
    ---------------------------------------------------
    
    ClientError: auth error: missing bearer token: {"response":{"errors":[{"message":"auth error: missing bearer token"}],"status":200,"headers":{}},"request":{"query":"{\n\t\t\tdexSwapFactories {\n\t\t\t\ttotalVolumeETH\n\t\t\t}\n\t\t}"}}
        at /home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/src/index.ts:498:11
        at step (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:67:23)
        at Object.next (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:48:53)
        at fulfilled (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/graphql-request/dist/index.js:39:58)
        at processTicksAndRejections (node:internal/process/task_queues:95:5) {
      response: {
        errors: [ [Object] ],
        status: 200,
        headers: Headers { [Symbol(map)]: [Object: null prototype] }
      },
      request: {
        query: '{\n\t\t\tdexSwapFactories {\n\t\t\t\ttotalVolumeETH\n\t\t\t}\n\t\t}',
        variables: undefined
      },
      chain: 'arbitrum'
    }

File: aggregators/dexswap/index.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { exportDexVolumeAndFees } from "../../helpers/dexVolumeLogs";
+import request, { gql } from "graphql-request";
+import { SimpleAdapter } from "../../adapters/types";
+import { CHAIN } from "../../helpers/chains";
+import BigNumber from "bignumber.js";
+
+const FACTORY_ADDRESS = '0x3e40739d8478c58f9b973266974c58998d4f9e8b';
+
+const apiKey = '';
+const graphUrl = `https://gateway-arbitrum.network.thegraph.com/api/${apiKey}/subgraphs/id/AaAYom6ATrw15xZq3LJijLHsBj8xWdcL11Xg6sCxEBqZ`;
```
Comment (dtmkeng) [inline aggregators/dexswap/index.ts:11]: you can use our function like this https://github.com/DefiLlama/dimension-adapters/blob/master/fees/thena-v3.ts#L20

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: aggregators/dexswap/index.ts (hunk 1 above)
Comment (tuco-gordian) [reply aggregators/dexswap/index.ts:11]: I don't understand how to use this function, can you explain it to me?

**Pull request merged**